core/templates/dev/head/pages/exploration_editor/history_tab/HistoryServices.js

```diff
@@ -198,12 +198,23 @@ oppia.factory('compareVersionsService', [
           v1Data: $http.get(explorationDataUrl + v1),
           v2Data: $http.get(explorationDataUrl + v2)
         }).then(function(response) {
-          var v1States = response.v1Data.data.states;
-          var v2States = response.v2Data.data.states;
+          var v1StatesDict = response.v1Data.data.states;
+          var v2StatesDict = response.v2Data.data.states;
 
           // Track changes from v1 to LCA, and then from LCA to v2.
           var lca = versionsTreeService.findLCA(v1, v2);
 
+          var v1States = {};
+          var v2States = {};
+          for (var stateName in v1StatesDict) {
```

Comment:
Doesn't need to be in this PR, but I was noticing that we're starting to use this sort of logic repeatedly. Do you think we ought to have a States domain object as well (with a built-in create() method)? It could be a fairly small object but it would be nice to encapsulate this logic.